src/displayapp/DisplayApp.cpp

```diff
@@ -141,6 +141,9 @@ void DisplayApp::Refresh() {
   if (messageReceived) {
     switch (msg) {
       case Messages::GoToSleep:
+        if (bleController.IsFirmwareUpdating()) {
```

Comment:
This extra precaution should not be necessary, as DfuService sends the message `Messages::BleFirmwareUpdateStarted` to SystemTask when it's starting the update procedure. 
When receiving this message, SystemTask sets the flag `doNotGoToSleep` to true so that it doesn't go to sleep until it receives the message `Messages::BleFirmwareUpdateFinished`.

Unless there's a weakness in this implementation?

Reply:
Some apps enable sleep on exit. If the firmware update is started while an app like this is running, the watch will go to sleep during the update. If this was fixed, then the check probably won't be needed. I didn't intend on this to be just a workaround for that though but just extra protection for any future changes.

Reply:
Mhm ok, I see what you mean : DfuService sends the message, SystemTask receives it, set the flash `doNotGoToSleep` to true and then sends a message to DisplayApp that closes the current app and open the FirmwareUpdate app.
If the app that was running re-enable sleep, the firmware update app will go to sleep after the timeout, yes.

So yes, this is an easy workaround, but I was thinking : shouldn't we send the message 'disable sleep' from the firmware update app ? This way, sleep mode will be disabled/enabled in the correct order. What do you think?

Reply:
I thought about that, but the reason I didn't do it was because the app isn't actually responsible for the update.

Maybe this check should instead be in systemtask message EnableSleeping?
```
case Messages::EnableSleeping:
  if (!bleController.IsFirmwareUpdating()) {
    doNotGoToSleep = false;
  }
  break;
```

Reply:
Well, sleeping can be considered as mostly a UI think... But sleep also means that the SPI memory will also be put to sleep, and it'll make the OTA fail so... it's not that simple, indeed.
Yes, we can also put this check the EnableSleeping, I don't know which one would be the best...

Reply:
I made the change I proposed as it addresses the existing issue more directly.

Reply:
Looks good to me, thanks! So this PR is ready to be merged?

Reply:
Yep, should be fine.